### PR TITLE
Add more UD domains

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/36131
Resolves https://github.com/brave/brave-browser/issues/29371
Resolves https://github.com/brave/brave-browser/issues/30290

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Set IPFS setting to gateway and visit these test domains.
reseller-test-udtesting-319708317921.altimist
reseller-test-udtesting-266798386399.anime
reseller-test-udtesting-257453775377.klever
reseller-test-udtesting-444498982797.manga
reseller-test-udtesting-260502680311.polygon
reseller-test-udtesting-451650328921.unstoppable

### CODE REVIEW

**Review by nuo-xu — APPROVED** [review]

**Review by darkdh — APPROVED** [review]

**Review by Douglashdaniel — APPROVED** [review]
++ Wallet Front-end

File: ios/brave-ios/Sources/BraveWallet/WalletConstants.swift [hunk 1]
```diff
@@ -100,7 +100,11 @@ public struct WalletConstants {
   /// The supported Solana Name Service (SNS) extensions
   static let supportedSNSExtensions = [".sol"]
   /// The supported Unstoppable Domain (UD) extensions
-  public static let supportedUDExtensions = [".crypto", ".x", ".nft", ".dao", ".wallet", ".blockchain", ".bitcoin", ".zil"]
+  public static let supportedUDExtensions = [
+    ".crypto", ".x", ".nft", ".dao", ".wallet",
+    ".blockchain", ".bitcoin", ".zil", ".altimist", ".anime",
+    ".klever", ".manga", ".polygon", ".unstoppable", ".pudgy"
+  ]
```
Comment (supermassive) [inline ios/brave-ios/Sources/BraveWallet/WalletConstants.swift:107]: I would say we need that list sorted and in one place. 
We should have a mojo method to backend `ResolveWalletAddress(string address, BlockhainToken token)` which is triggered when address field has a dot.
Not now of course :)

Comment (StephenHeaps) [reply ios/brave-ios/Sources/BraveWallet/WalletConstants.swift:107]: ++ for both

A few spots we need to expose constants better for all platforms, like currency codes with [#22481](https://github.com/brave/brave-browser/issues/22481) :)

**Review by supermassive — APPROVED** [review]

**Review by kdenhartog — APPROVED** [review]
LGTM for security

**Review by iefremov — APPROVED** [review]

**Review by mkarolin — APPROVED** [review]
LGTM, but if we are going to keep adding/modifying these, perhaps we should consider changing the UI so that these are split out into a separate string that doesn't need to be translated.

**Comment by yrliou** [comment]
@mkarolin Totally agreed with you, this one has some urgency since we need to uplift to 1.63.x, I'll create a follow-up issue to fix this on master.

File: ios/brave-ios/Sources/BraveWallet/WalletConstants.swift (hunk 1 above)
Comment (StephenHeaps) [reply ios/brave-ios/Sources/BraveWallet/WalletConstants.swift:107]: 1. https://github.com/brave/brave-browser/issues/36165
2. https://github.com/brave/brave-browser/issues/26655

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.65.8 Chromium: 122.0.6261.43 (Official Build) nightly (64-bit)
-- | --
Revision | 99fa7076dd9d262e50bfea1be951e68b0ce236d0
OS | Windows 11 Version 23H2 (Build 22631.3155)
```

Using the STR/Cases outlined via https://github.com/brave/brave-core/pull/22102#issue-2137328755 and ensured that the `UD` domains resolved without any issues after `Gateway` was enabled via `Method to resolve IPFS resources` under `brave://settings/web3` as per the following:

**`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|----------------|----------------|----------------|----------------|----------------
UD1 | UD2 | UD3 | UD4 | UD5 | UD6 | UD7

**Pull request merged**